src/Data/Text/IO.hs

```diff
@@ -101,13 +101,35 @@ readChunkEof :: Handle__ -> CharBuffer -> IO (Text, Bool)
 readChunkEof hh buf = do t <- readChunk hh buf
                          return (t, False)
 
--- | /Experimental./ Read a single chunk of strict text from a
+-- | Read a single chunk of strict text from a
 -- 'Handle'. The size of the chunk depends on the amount of input
 -- currently buffered.
 --
 -- This function blocks only if there is no data available, and EOF
 -- has not yet been reached. Once EOF is reached, this function
 -- returns an empty string instead of throwing an exception.
+--
+-- == Behavior
+--
+-- Unlike byte-oriented functions, 'hGetChunk' operates on complete UTF-8
+-- characters. Since UTF-8 characters can occupy 1 to 4 bytes, this function
+-- cannot guarantee reading an exact number of bytes. Instead, it reads
+-- complete characters up to the handle's internal buffer limit.
+--
+-- == Buffer Size
+--
+-- The maximum chunk size is determined by the handle's internal character
+-- buffer, which is set to 2048 characters (not bytes) by the GHC runtime
```

Comment:
The characters for `dEFAULT_CHAR_BUFFER_SIZE` are `char` from C and each of them takes exactly 4 bytes. See https://hackage.haskell.org/package/ghc-internal-9.1001.0/docs/GHC-Internal-IO-Buffer.html#v:newCharBuffer and https://hackage.haskell.org/package/ghc-internal-9.1001.0/docs/GHC-Internal-IO-Buffer.html#v:charSize.

So I'd rather say that the default character buffer is 8192 bytes long.

Reply:
I thought that just decoded `Char`s before writing them into the `CharBuffer` character by character, so that it doesn't matter how a `CharBuffer` represents them internally?

Reply:
Mm, yeah, that's probably fair.